File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/IntradayBarRequest/RequestIntradayBarElementInt.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.IntradayBarRequest.RequestIntradayBarElementInt.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.IntradayBarRequest;

public class RequestIntradayBarElementInt extends RequestIntradayBarElementString
{
    private final int _value;

    RequestIntradayBarElementInt(String elementName, int value)
    {
    	super(elementName, String.valueOf(value));
        this._value = value;
    }

    //I can't override GetElementAsInt32 here because the Bloomberg Request object stores ints as strings, not ints.  You can't convert the string to an int
    protected int getInt()
    {
    	return this._value;
    }
}

File: BEmu_cpp/src/IntradayBarRequest/RequestIntradayBarElementInt.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayBarRequest/RequestIntradayBarElementInt.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "BloombergTypes/Name.h"
#include "BloombergTypes/ElementPtr.h"
#include "IntradayBarRequest/RequestIntradayBarElementInt.h"

namespace BEmu
{
	namespace IntradayBarRequest
	{
		RequestIntradayBarElementInt::RequestIntradayBarElementInt(const std::string& elementName, int ivalue) : RequestIntradayBarElementString(elementName, "")
		{
			std::stringstream ss;
			ss << ivalue;
			this->_value = ss.str();

			this->_ivalue = ivalue;
		}

		RequestIntradayBarElementInt::~RequestIntradayBarElementInt()
		{
		}

		int RequestIntradayBarElementInt::getInt() const
		{
			return this->_ivalue;
		}

	}
}